src/Model/Notification.php

```diff
@@ -154,4 +157,160 @@ public static function getType(array $notification): string
 
 		return $type;
 	}
+
+	/**
+	 * Create a notification message for the given notification
+	 *
+	 * @param array $notification
+	 * @return array with the elements "causer", "notification", "plain" and "rich"
+	 */
+	public static function getMessage(array $notification)
```

Comment:
Maybe this is more a `Repository` thing than a `Model`, but it's up to @MrPetovan , because I don't know the last state of how we wan't to use the classes.

Reply:
I'm really unsure about the whole class structure here.

Reply:
The `Notification` Model class is an outlier as it was the start of the conversation about Domain Driven Design and concepts like Factories, Repositories, etc...

Now it's clearer in my head but this class still exist as an in-between state. It would need significant refactoring to match the current paradigm that I don't think fits the scope of this PR. Also I probably should do it myself to match the current paradigm since it hasn't been formalized and nobody else but me seems to have a good idea of it. 😅

Reply:
More to the point of this particular method, if the array is supposed to be a row from the `notify` table, then `$this->{field}` should be used instead of `$notification[{field}]`.

Reply:
Also there is the problem that this class is about the "notify" table which will eventually will be replaced by the "notification" table that already is used for the notifications in the API. So hopefully at some point in time not only the "enotify.php" will re completely removed but also we will need a "notification" class that handles the "notification" table. And since this is a work in progress we will need a solution for the time when both tables are in use.

Reply:
@MrPetovan if you do have time, I'd love if you could describe your thoughts at the dev-board :-) .. I guess we do use the model instances as  "Data Transfer Objects", means that they should not contain "business logic" per se but rather be for accessing data, but I'm not sure ..

Reply:
What's the "dev-board"? I believe I already described my final thoughts on the subject in the Friendica Developers Forum, I can't find it anymore, but I can start again if I have to.